### PR TITLE
Lessons 2-3

### PR DESCRIPTION
**Pull request by buttsjames**

### CODE REVIEW

**Comment by buttsjames** [comment]
Bringing in Lesson02 and Lesson03

File: students/jbutts/Lesson03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+show_list("Here's the fruits before", fruits)
+fruits = stiurf
+show_list("Here's the same fruits spelled backwards", fruits)
+
+print("\n\nB:\n\n")
+fruits_orig.pop(len(fruits_orig)-1)
+
+show_list("here's the original list with the last item removed", fruits_orig)
+show_list("here's a copy the original list", fruits_copy)
```
Comment (charlesrobison) [inline students/jbutts/Lesson03/list_lab.py]: Great job on this!

File: students/jbutts/Lesson03/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                print(format_email(donor_name, amount, donors))
+                break
+        else:
+            response = str(input("Donor {} doesn't exist, add them (y/n)?:".format(donor_name)))
+            if response.lower() == 'y':
+                add_contribution(donor_name, amount, donors)
+                print(format_email(donor_name, amount, donors))
+
+
+
```
Comment (charlesrobison) [inline students/jbutts/Lesson03/mailroom.py]: Lots of extra lines here - go ahead and remove

File: students/jbutts/Lesson03/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        elif response == 2:
+            show_donors(donor_db)
+
+
+
+if __name__ == '__main__':
+    # We're not getting imported, run main():
+    donor_db = []
+    donor_db = populate_data(donor_db)
+    main()
```
Comment (charlesrobison) [inline students/jbutts/Lesson03/mailroom.py]: @buttsjames - note you're missing the sorting requirement for this assignment: "If the user (you) selected “Create a Report,” print a list of your donors, sorted by total historical donation amount."

**Comment by buttsjames** [comment]
Hi,

I'm getting everything up-to-date and will submit through lesson5 with your
feedback incorporated.

I had some issues with Git (accidentally committed a very large file as
part of the file lab, and had to do a bunch of stuff to delete it). My
latest pull request look like it has the right files, but also has a large
number of commits from other people in the class.

https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/209/commits

I'm not sure if this is normal, but I suspect I messed something up. Any
feedback on what I need to do to get this right?

Thanks,

James

**Pull request merged**